doc-deploy-dev/action.yml

```diff
@@ -20,123 +30,145 @@ inputs:
+  commit-message:
+    description: "Commit message used when deploying the documentation"
     required: false
-    default: true
+    default: 'DOC: update development documentation'
     type: string
 
+
+
 runs:
   using: "composite"
   steps:
 
-    - name: "Install system dependencies"
+    # Checkout the repository branch for deploying the documentation. If this
+    # step fails, then it means that the provided token is not valid.
+
+    - name: "Get the name of the repository"
       shell: bash
       run: |
-        sudo apt-get install cargo curl
-        cargo install ouch && ouch --version
+        if [[ "${{ inputs.repository }}" == "current" ]]; then
+          echo "REPOSITORY=${{ github.repository }}" >> $GITHUB_ENV
+        else
+          echo "REPOSITORY=${{ inputs.repository }}" >> $GITHUB_ENV
+        fi
 
-    - name: "Checkout project in the GitHub Pages branch"
+    - name: "Checkout ${{ env.REPOSITORY }} repository"
       uses: actions/checkout@v3
-      if: inputs.checkout == 'true'
       with:
-        ref: 'gh-pages'
+        repository: ${{ env.REPOSITORY }}
+        ref: ${{ inputs.branch }}
+        token: ${{ inputs.token }}
 
-    - name: "Create a fresh development folder"
-      shell: bash
-      run: |
-        rm -rf dev/ && mkdir dev/
+    # Downlaod the documentation artifact from the current workflow. If the
+    # artifact contains a compressed file, decompress it. Display the structure
+    # of the 'version/dev' directory at the end of the process to verify the
+    # layout of the folder is the right one.
 
-    - name: "Download the HTML documentation artifact from desired workflow"
-      uses: dawidd6/action-download-artifact@v2
-      if: inputs.workflow != ''
-      with:
-        github_token: ${{ inputs.token }}
-        name: ${{ inputs.doc-artifact-name }}
-        path: dev
-        workflow: ${{ inputs.workflow }}
-        workflow_conclusion: ${{ inputs.workflow-conclusion }}
+    - name: "Clean destination folder"
+      shell: bash
+      run:
+        rm -rf version/dev && mkdir -p version/dev
 
-    - name: "Download the HTML documentation artifact from current workflow"
+    - name: "Download the development documentation artifact"
       uses: actions/download-artifact@v3
-      if: inputs.workflow == ''
       with:
         name: ${{ inputs.doc-artifact-name }}
-        path: dev
+        path: version/dev
 
-    - name: "Decompress artifact content if required"
+    - name: "Decompress artifact content"
       shell: bash
-      if: inputs.decompress_artifact == 'true'
-      working-directory: dev
+      if: inputs.decompress-artifact == 'true'
       run: |
-        compressed_artifact=$(ls .)
-        ouch decompress $compressed_artifact
-        rm $compressed_artifact
+        sudo apt-get install cargo && cargo install ouch && ouch --version
+        cd version/dev && compressed_artifact=$(ls .)
+        ouch decompress $compressed_artifact && rm $compressed_artifact
```

Comment:
This package is great 😄 even reading the CLI is funny 😄 

![image](https://user-images.githubusercontent.com/37798125/218967657-95d7a243-d9a7-4b0b-bc46-d8235165fdf5.png)